edgelet/cert-client/src/error.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::fmt;
+use std::fmt::Display;
+use std::str;
+
+use failure::{Backtrace, Context, Fail};
+use hyper::StatusCode;
+
+#[derive(Debug)]
+pub struct Error {
+    inner: Context<ErrorKind>,
+}
+
+#[derive(Debug, Fail)]
+pub enum ErrorKind {
```

Comment:
You're using `Fail`, so the variants are the operation context. One variant for each operation.

```rust
#[derive(Debug, Fail)]
pub enum ErrorKind {
    #[fail(display = "could not delete cert with ID {}", _0)]
    DeleteCert(String),

    #[fail(display = "could not get cert with ID {}", _0)]
    GetCert(String),

    ...
}
```

ie rename `RequestType` to `ErrorKind` and add the ID etc data to the variants.

Reply:
I cleant up the RequestTypes. I will take another pass at error handling in my next PR.

Reply:
The callers are logging the operation type. I have set the context in refresh_cert instead of the wrong OperationType that was previously there.